packages/core/src/sessionInfo/SessionInfoManager.ts

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+/**
+ * @hidden
+ * @packageDocumentation
+ */
+
+import { v4 } from "uuid";
+import { fetch } from "@inrupt/universal-fetch";
+import type {
+  ISessionInfo,
+  ISessionInfoManager,
+  ISessionInternalInfo,
+  ISessionInfoManagerOptions,
+  IStorageUtility,
+} from "..";
+
+export function getUnauthenticatedSession(): ISessionInfo & {
+  fetch: typeof fetch;
+} {
+  return {
+    isLoggedIn: false,
+    sessionId: v4(),
+    fetch,
+  };
+}
+
+/**
+ * @param sessionId
+ * @param storage
+ * @hidden
+ */
+export async function clear(
+  sessionId: string,
+  storage: IStorageUtility
+): Promise<void> {
+  await Promise.all([
+    storage.deleteAllUserData(sessionId, { secure: false }),
+    storage.deleteAllUserData(sessionId, { secure: true }),
+    // FIXME: This is needed until the DPoP key is stored safely
+    storage.delete("clientKey", { secure: false }),
```

Comment:
This is no longer necessary, the DPoP key isn't stored outside of the fetch scope.